src/routes/settings.tsx

```diff
@@ -0,0 +1,414 @@
+          headers: {
+            authorization: `Bearer ${auth?.tokenRaw}`,
+          },
+          body: JSON.stringify({
+            path: `${file.name}`,
+            contentType: file.type,
+          }),
+        })
+      ).json();
+
+      const uploadResponse = await fetch(presignedUrl.data, {
+        method: "PUT",
+        headers: {
+          "Content-Type": file.type,
+        },
+        body: file,
+      });
+
+      if (uploadResponse.status !== 200) {
+        throw new Error();
+      }
+      getLogosList();
+      handleLogoSelection(file.name);
+    } catch (error) {
+      console.error(t("errorHandling.errorUploadingImage"), error);
+    }
+
+    setLoading(false);
+  };
+
+  /**
+   * Disable project theme handler
+   */
+  const disableProjectThemeHandler = () => {
+    handleProjectThemeChange(DEFAULT_THEME_COLORS[0].value, DEFAULT_LOGO);
+  };
+
+  /**
+   * Project theme change handler
+   *
+   * @param color string
+   * @param logo string
+   */
+  const handleProjectThemeChange = (color?: string, logo?: string) => {
+    if (!selectedProject) {
+      setSelectedColor(color ?? selectedColor);
+      setSelectedLogo(logo ?? selectedLogo);
+      return;
+    }
+
+    if (projectThemes.length) {
+      updateProjectTheme(selectedProject, color, logo);
+    } else {
+      createProjectTheme(selectedProject, color, logo);
+    }
+  };
+
+  /**
+   * Handles color selection
+   *
+   * @param color string
+   */
+  const handleColorSelection = (color: string) => {
+    handleProjectThemeChange(color);
+  };
+
+  /**
+   * Handles custom color selection
+   *
+   * @param color string
+   */
+  const handleCustomColorSelection = (color: string) => {
+    setSelectedColor(color);
+    if (debounceTimeoutRef.current) {
+      clearTimeout(debounceTimeoutRef.current);
+    }
+    debounceTimeoutRef.current = setTimeout(() => {
+      handleProjectThemeChange(color);
+    }, 4000);
+  };
+
+  /**
+   * Handles logo selection
+   */
+  const handleLogoSelection = (logo: string) => {
+    handleProjectThemeChange(undefined, logo);
+  };
+
+  /**
+   * Handles project selection
+   *
+   * @param event event
+   */
+  const handleProjectSelection = ({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+    setSelectedProject(value);
+  };
+
+  /**
+   * Renders color buttons
+   */
+  const renderColorsButtons = () =>
+    DEFAULT_THEME_COLORS.map((color) => (
+      <Box
+        key={color.name}
+        sx={{
+          display: "flex",
+          flexDirection: "column",
+          alignSelf: "flex-start",
+          alignItems: "center",
+          textAlign: "center",
+          width: "10%",
+        }}
+        onClick={() => handleColorSelection(color.value)}
+      >
+        <Button
+          variant="contained"
+          sx={{
+            backgroundColor: color.value,
+            minWidth: 0,
+            width: "2.5rem",
+            height: "2.5rem",
+            borderRadius: "50%, ",
```

Comment:
Is this correct? Also, border radius is not great with 50%, as if the element is not square, the borders looks bent. Use e.g. 9999px instead.

Reply:
Looks like a typo, changed as you suggest